### PR TITLE
Use k8s.io/klog/v2 for logging

### LINKED ISSUE #7: Support verbose/granular output levels
As discussed in #6 , `cli` should output via `klog` and support multiple levels of verbosity.

### PR DESCRIPTION
**Pull request by coreydaley**
Screenshot from 2021-04-21 09-06-51
Replace fmt.Printf statements with klog.Infof, etc

Fixes #7

### CODE REVIEW

**Comment by coreydaley** [comment]
/assign @adambkaplan 
cc: @alicerum

File: pkg/shp/cmd/build/run.go [hunk 1]
```diff
@@ -71,6 +71,6 @@ func (c *RunCommand) Run(params *params.Params, ioStreams *genericclioptions.IOS
 		return err
 	}
 
-	fmt.Fprintf(ioStreams.Out, "BuildRun created %q\n", buildRun.Name)
+	klog.Infof("BuildRun created %q\n", buildRun.Name)
```
Comment (adambkaplan) [inline pkg/shp/cmd/build/run.go:74]: Grammar nit
```suggestion
	klog.Infof("Created buildrun %q\n", buildRun.Name)
```

File: pkg/shp/cmd/buildrun/delete.go [hunk 2]
```diff
@@ -54,7 +53,7 @@ func (c *DeleteCommand) Run(params *params.Params, ioStreams *genericclioptions.
 		return err
 	}
 
-	fmt.Fprintf(ioStreams.Out, "BuildRun deleted '%v'\n", c.name)
+	klog.Infof("BuildRun deleted '%v'\n", c.name)
```
Comment (adambkaplan) [inline pkg/shp/cmd/buildrun/delete.go:56]:
```suggestion
	klog.Infof("Deleted buildrun '%v'\n", c.name)
```

File: pkg/shp/cmd/buildrun/logs.go [hunk 3]
```diff
@@ -73,7 +74,7 @@ func (c *LogsCommand) Run(params *params.Params, ioStreams *genericclioptions.IO
 		return fmt.Errorf("no builder pod found for BuildRun %q", c.name)
 	}
 
-	fmt.Fprintf(ioStreams.Out, "Obtaining logs for BuildRun %q\n\n", c.name)
+	klog.Infof("Obtaining logs for BuildRun %q\n\n", c.name)
```
Comment (adambkaplan) [inline pkg/shp/cmd/buildrun/logs.go:77]: Do we need this printed to stdout? `kubectl logs` doesn't print something like this unless debug/verbose logging is turned on.

Perhaps log at V(5) instead?

**Review by adambkaplan — CHANGES_REQUESTED** [review]
Few nits from the initial implementation.

Does klog print to stdout/stderr by default? A screenshot of the logging via klog would be helpful.

File: cmd/shp/main.go [hunk 4]
```diff
@@ -15,13 +14,13 @@ import (
...
-	flags := pflag.NewFlagSet(ApplicationName, pflag.ExitOnError)
-	pflag.CommandLine = flags
+	klogFlags := flag.NewFlagSet(ApplicationName, flag.ExitOnError)
+	klog.InitFlags(klogFlags)
 
 	streams := genericclioptions.IOStreams{In: os.Stdin, Out: os.Stdout, ErrOut: os.Stderr}
 	rootCmd := cmd.NewCmdSHP(&streams)
 	if err := rootCmd.Execute(); err != nil {
-		fmt.Fprintf(os.Stderr, "ERROR: %v\n", err)
+		klog.Errorf("ERROR: %v\n", err)
```
[the remark this answers is not in the document]
Comment (coreydaley) [reply cmd/shp/main.go:23]: fixed in next push

File: pkg/shp/cmd/build/create.go [hunk 5]
```diff
@@ -101,6 +101,6 @@ func (sc *CreateCommand) Run(params *params.Params, io *genericclioptions.IOStre
 		return err
 	}
 
-	fmt.Fprintf(io.Out, "Build created %q\n", sc.name)
+	klog.Infof("Build created %q\n", sc.name)
```
[the remark this answers is not in the document]
Comment (coreydaley) [reply pkg/shp/cmd/build/create.go:104]: fixed in next push

File: pkg/shp/cmd/build/delete.go [hunk 6]
```diff
@@ -73,12 +74,12 @@ func (c *DeleteCommand) Run(params *params.Params, io *genericclioptions.IOStrea
...
 		for _, buildrun := range brList.Items {
 			if err := brr.Delete(c.cmd.Context(), buildrun.Name); err != nil {
-				fmt.Fprintf(io.ErrOut, "Error deleting BuildRun %q: %v\n", buildrun.Name, err)
+				klog.Errorf("Error deleting BuildRun %q: %v\n", buildrun.Name, err)
 			}
 		}
 	}
 
-	fmt.Fprintf(io.Out, "Build deleted %q\n", c.name)
+	klog.Infof("Build deleted %q\n", c.name)
```
[the remark this answers is not in the document]
Comment (coreydaley) [reply pkg/shp/cmd/build/delete.go:82]: fixed in next push

File: pkg/shp/cmd/build/run.go (hunk 1 above)
Comment (coreydaley) [reply pkg/shp/cmd/build/run.go:74]: fixed in next push

File: pkg/shp/cmd/buildrun/delete.go (hunk 2 above)
Comment (coreydaley) [reply pkg/shp/cmd/buildrun/delete.go:56]: fixed in next push

File: pkg/shp/cmd/buildrun/logs.go (hunk 3 above)
Comment (coreydaley) [reply pkg/shp/cmd/buildrun/logs.go:77]: fixed in next push

**Comment by coreydaley** [comment]
@adambkaplan ptal, all comments addressed

**Comment by adambkaplan** [comment]
screen shot if possible?

**Comment by gabemontero** [comment]
/hold
for @adambkaplan 's screen shot

/lgtm

**Comment by coreydaley** [comment]
> screen shot if possible?

Added to original post

**Comment by adambkaplan** [comment]
I don't like the extra info that klog prints - it clutters display and gets in the way of the message presented to an end user. The extra info is more appropriate for server logs, where that metadata can be used to debug issues.

**Comment by gabemontero** [comment]
> I don't like the extra info that klog prints - it clutters display and gets in the way of the message presented to an end user. The extra info is more appropriate for server logs, where that metadata can be used to debug issues.

fwiw openshift/oc and underlying kubectl uses klog v2

now, there are places with more complex displaying, like `explain`, where they use PrintWriters and the like

I wonder if a similar mixture will end up being appropriate for this one

Also by comparison, tkn from tekton seems less beholden to klog .... perhaps that should be the guide.

We should model after one of the existing "stakes in the ground" IMO.

**Pull request closed without merging**